### DISCUSSION TITLE
"up-accept-location" I cannot make it work [Solved]

### DISCUSSION THREAD

**Discussion by manlioesp** [unpoly/unpoly#283] [General]
Hi guys, i'm having problem with up-accept-location and probably I don't understand it correctly due to my low skill.
I'm making a subinteraction layer like the DEMO company and projects CRUD.
Basically I'm at this point https://demo.unpoly.com/companies/113984 and I'm using this code for the button

`<a href="<?=$pages->get("133200")->url?>?action=save" up-layer="new"  up-accept-location="<?=$pages->get("133200")->url?>?action=save" up-on-accepted="up.reload('.table')">Add project </a>`

From what I have understood, as soon as the new layer reach the page "<?=$pages->get("133200")->url?>?action=save" unploy should close the modal and return to parent layer, but it is not working for me.
The page reached by this button is more or less like this 

(Github is stripping some code so I attach a reference image)

![screen](https://user-images.githubusercontent.com/9567588/133978933-06835704-5426-4fa6-9bd0-09a610c14059.jpg)

What I'm missing?
Thank you for any help.

**Comment by adam12** [comment]
Every page between you clicking the `Add project` button and the accept location page _must_ be handled by Unpoly. Can you confirm that each page is done using `[up-follow]`, `[up-submit]`, or similar attribute?

Using `up.log.enable()` in the console might help as well.

**Comment by triskweline** [comment]
Also check if you're actually reaching the acceptance URL. The log should have a line like this:

```
[up:location:changed] Advanced to location /path-that-will-accept-the-modal
```

By default Unpoly will only update history for a GET request. If you're making a POST request to save the form, the history will not be changed. The demo redirects to the target URL for this purpose. If you can't do this in your code you may want to use `[up-accept-event]` instead.

**Reply by manlioesp** [reply]
Thank you for your quick replies and sorry for my late reply. I was able to make `[up-accept location]` work sending the form with GET request adding also a  `up-history="true"`. 
I would like to use the `[up-accept-event]` and maintain the POST request, but I wasn't able to make it to work.
How can I trigger the event after the save record code?
Instead of `up-accept-location="url..."` I used `up-accept-event="confirm:event"` and then I tried to trigger the event with a
`<body onload="up.emit('confirm:event', { foo: 'bar' });">`  inserted in the page that save the data with PHP  (I know I doesn't need foo:'bar', but I have used this code for testing).
Unluckily it doesn't wotk. Can you help me with some code example?
Thank you very much. Have a nice day.

**Reply by adam12** [reply]
You should be emitting on the server, by sending the appropriate header on that page.

```
X-Up-Events: [{ "type": "confirm:event", "foo": "bar" }]
```

Perhaps something like this? (I don't use PHP and cannot test it).

```php
header('X-Up-Events: [{ "type": "confirm:event", "foo": "bar" }]');
```

The documentation for `X-Up-Events` is [here](https://unpoly.com/X-Up-Events).

**Reply by manlioesp** [reply]
Thank you very much for your patience Adam. Finally I got it working, Unpoly is amazing.
 I had to add "layer":"current" to make it works, and below there is the php code I have used if someone else is interested.

```
header('X-Up-Events: [{ "type": "confirm","layer": "current"}]');
echo json_encode(['location' => $page->url]);
exit;
```



I found a small typo in the documentation.
At this link https://unpoly.com/X-Up-Events at the end of the page you find
X-Up-Events: [{ "type": "user:created", **"name:"** "foobar", "layer": "current" }] 

It should be **"name":**
